submarine-workbench/workbench-server/src/main/java/org/apache/submarine/server/WorkbenchGitHubServer.java

```diff
@@ -0,0 +1,194 @@
+   */
+  public void commit(String localPath, String message) throws IOException, GitAPIException {
+    // Git repository address
+    try (Git git = new Git(new FileRepository(localPath + ".git"))) {
+      // Submit code
+      git.commit().setMessage(message).call();
+    }
+  }
+
+  /**
+   * To execute push command
+   * @param localPath
+   * @param token
+   * @throws IOException
+   * @throws GitAPIException
+   */
+  public void push(String localPath, String token, String remote) throws IOException, GitAPIException {
+    CredentialsProvider credentialsProvider =
+        new UsernamePasswordCredentialsProvider("PRIVATE-TOKEN", token);
+    // Git repository address
+    try (Git git = new Git(new FileRepository(localPath + ".git"))) {
+      git.push().setRemote(remote).setCredentialsProvider(credentialsProvider).call();
+    }
+  }
+
+  /**
+   * To execute branchCreate command
+   * @param localPath
+   * @param branchName
+   * @throws IOException
+   * @throws GitAPIException
+   */
+  public void branchCreate(String localPath, String branchName) throws IOException, GitAPIException {
+    try (Git git = new Git(new FileRepository(localPath + ".git"))) {
+      ListBranchCommand listBranchCommand = git.branchList();
+      List<Ref> list = listBranchCommand.call();
+      boolean existsBranch = false;
+      for (Ref ref : list) {
+        if (ref.getName().endsWith(branchName)) {
+          existsBranch = true;
+          break;
+        }
+      }
+      if (!existsBranch) {
+        // Create branch
+        git.branchCreate().setName(branchName).call();
+      } else {
+        LOG.warn("{} already exists.", branchName);
+      }
+    }
+  }
+
+  /**
+   * To execute checkout command
+   * @param localPath
+   * @param branchName
+   * @throws IOException
+   * @throws GitAPIException
+   */
+  public void checkout(String localPath, String branchName) throws IOException, GitAPIException {
+    try (Git git = new Git(new FileRepository(localPath + ".git"))) {
+      git.checkout().setName(branchName).call();
+    }
+  }
+
+  /**
+   * To execute rebase command
+   * @param localPath
+   * @param branchName
+   * @param upstreamName
+   * @throws IOException
+   * @throws GitAPIException
+   */
+  public void rebase(String localPath, String branchName, String upstreamName)
+      throws IOException, GitAPIException {
+    try (Git git = new Git(new FileRepository(localPath + ".git"))) {
+      git.rebase().setUpstream(branchName).setUpstreamName(upstreamName).call();
+    }
+  }
+
+  public static void main(String[] args) {
+    WorkbenchGitHubServer workbenchGitHubServer = new WorkbenchGitHubServer();
+    String localPath = "F:\\zhulinhao\\test2\\";
+    try {
+      // workbenchGitHubServer.branchCreate(localPath, "branch-2");
+      // workbenchGitHubServer.checkout(localPath, "branch-2");
+      workbenchGitHubServer.rebase(localPath, "master", "upstream");
+    } catch (IOException e) {
+      e.printStackTrace();
```

Comment:
Change `e.printStackTrace();` to `LOG.error(e.getMeesage(), e)`

Reply:
Done.